RELEASE.md

```diff
@@ -0,0 +1,94 @@
+# Rusk Monorepo Release Process
+
+## **1. Branching Strategy**
+
+### `master` Branch
+- **Purpose:** Active development.
+- **Policy:**
+  - All new features behind feature flags.
+  - Merged only after passing CI, integration tests, and approval.
+
+### `devnet` Branch
+- **Purpose:** Weekly snapshots from `master` for early testing.
+- **Policy:** Updated weekly, deploying directly to the Devnet environment.
+
+### Release Branches (`rusk-release-vX.Y`)
+- **Purpose:** Stable releases for mainnet or testnet deployments.
+- **Why We Need It:**
+  - **Hotfix Management:** Allows hotfixes to be applied without branching from tags.
+  - **Long-Term Support:** Enables long-term support and backporting of fixes.
+  - **Clear Release Tracking:** Provides a clear path for version progression.
+
+- **Creation:** Every 1-3 months or as needed.
+- **Hotfixes:** Created directly on the (affected) release branch.
+
+---
+
+## **2. Tagging and Release Strategy**
+
+### Libraries
+- `dusk-abi-vX.Y.Z` (rusk-framework release?)
+- `dusk-core-vX.Y.Z` (rusk-framework release?)
+- `dusk-wallet-core-vX.Y.Z` (rusk-framework release?)
+- `w3sper.js-vX.Y.Z`
+
+### User-Facing Apps
+- `explorer-vX.Y.Z`
+- `rusk-wallet-vX.Y.Z`
+- `web-wallet-vX.Y.Z`
+
+### Blockchain Node
+- `rusk-vX.Y.Z` (node binary release)
+- `rusk-framework-vX.Y.Z` (library releases for components such as ABI, core, wallet-core)
+
+**Example:** `rusk-v1.2.3`
+
+---
+
+## **3. Release Lifecycle**
+
+1. **Day 0:** Create release branch `rusk-release-vX.Y` from `master`.
+2. **Day 1:** Deploy release to **Testnet**.
+3. **Day 7:** Release validated and stabilized in Testnet.
+4. **Day 10:** Finalize integration tests and prepare for mainnet deployment.
+5. **Day 14:** Community announcement and gradual provisioners upgrade.
+6. **Day 16:** Deploy `rusk-vX.Y.Z` to **Mainnet**.
```

Comment:
Isn't "provisioner upgrade" the same of "deploy to mainnet"?

What's the difference?

Reply:
> Isn't "provisioner upgrade" the same of "deploy to mainnet"?
> 
> What's the difference?

People already deploying the binary, but thinking about it it doesn't make a whole lot of sense since we'll specify an upgrade is active after a given block height into the future anyway if it's a protocol change. Maybe it makes sense to specify a minimum amount of time/blocks a protocol change needs to be activated.

In terms of the dates mention, they're just examples. Feel free to suggest a different time frame.